abi-layout.html

```diff
@@ -3346,7 +3346,27 @@ <h4> Compression </h4>
    &lt;substitution> ::= Si # ::std::basic_istream&lt;char,  std::char_traits&lt;char> >
    &lt;substitution> ::= So # ::std::basic_ostream&lt;char,  std::char_traits&lt;char> >
    &lt;substitution> ::= Sd # ::std::basic_iostream&lt;char, std::char_traits&lt;char> >
+</pre></font></code>
+
+To account for versioned standard library namespaces, the following substitutions
+are defined, where &lt;inline-ns> is the numeric part of a namespace name matching
+__[0-9]+, except __1 for which there is no substitution.
 
+<p><i>
+Note that versioning namespaces matching __1 are not compressed because that would
+constitute an ABI break for existing implementations, which already use that scheme.
+</i></p>
+
+<pre><font color=blue><code>
+   &lt;substitution> ::= Stv&lt;inline-ns>v # ::std::__[0-9]+
+   &lt;substitution> ::= Sav&lt;inline-ns>v # ::std::__[0-9]+::allocator
+   &lt;substitution> ::= Sbv&lt;inline-ns>v # ::std::__[0-9]+::basic_string
+   &lt;substitution> ::= Ssv&lt;inline-ns>v # ::std::__[0-9]+::basic_string &lt; char,
+                                                                        ::std::__[0-9]+::char_traits&lt;char>,
+                                                                        ::std::__[0-9]+::allocator&lt;char> >
+   &lt;substitution> ::= Siv&lt;inline-ns>v # ::std::__[0-9]+::basic_istream&lt;char,  std::__[0-9]+::char_traits&lt;char> >
+   &lt;substitution> ::= Sov&lt;inline-ns>v # ::std::__[0-9]+::basic_ostream&lt;char,  std::__[0-9]+::char_traits&lt;char> >
+   &lt;substitution> ::= Sdv&lt;inline-ns>v # ::std::__[0-9]+::basic_iostream&lt;char, std::__[0-9]+::char_traits&lt;char> >
```

Comment:
Optional suffices like turning `Ss` into `Ssv0v` don't work in the mangling grammar; types have to be self-limiting or else you get ambiguities.  For example, you want `_Z3fooISsv1vEv` to demangle to `foo<std::__1::string>()`, but it already demangles to `foo<std::string,void,v>()`.  You have to move this earlier in the production, like `Sv0_s`.